backup-stores/azure/src/main/java/io/camunda/zeebe/backup/azure/AzureBackupStore.java

```diff
@@ -108,7 +110,21 @@ public CompletableFuture<Collection<BackupStatus>> list(final BackupIdentifierWi
 
   @Override
   public CompletableFuture<Void> delete(final BackupIdentifier id) {
-    throw new UnsupportedOperationException();
+    return CompletableFuture.runAsync(
+        () -> {
+          final Manifest manifest = manifestManager.getManifest(id);
+          if (manifest == null) {
+            return;
+          } else if (manifest.statusCode() == StatusCode.IN_PROGRESS) {
+            throw new UnexpectedManifestState(
+                "Cannot delete Backup %s while saving is in progress.".formatted(id.toString()));
+          }
```

Comment:
:wrench: This feels like something the manifest manager should do. This would also have the advantage that the manifest manager can check that the manifest doesn't change between checking for in-progress and deleting (via the etag, like we did before).

Reply:
Its a good idea to move to the manifest manager 👍 , but do we need to check for the etags in the case of delete?
If it is in progress, then we don't delete.
if it is completed, then it wont be in progress again so we can delete, even if it gets marked as failed in the meantime.

Reply:
Yeah I think you are right that there are no cases right now where this would be necessary :+1: